pkg/controller/operators/catalog/step.go

```diff
@@ -75,17 +73,15 @@ func (b *builder) create(step v1alpha1.Step) (Stepper, error) {
 
 	switch step.Resource.Kind {
 	case crdKind:
-		version, err := crdlib.Version(&manifest)
+		groupVersion, err := crdlib.GroupVersion(&manifest)
 		if err != nil {
 			return nil, err
 		}
-
-		switch version {
-		case crdlib.V1Version:
-			return b.NewCRDV1Step(b.opclient.ApiextensionsInterface().ApiextensionsV1(), &step, manifest), nil
-		case crdlib.V1Beta1Version:
-			return b.NewCRDV1Beta1Step(b.opclient.ApiextensionsInterface().ApiextensionsV1beta1(), &step, manifest), nil
+		// v1beta1 CRDs are deprecated -- OLM no longer supports installing them
+		if deprecated(groupVersion) {
+			return nil, deprecatedError{k: crdKind, gv: groupVersion, name: step.Resource.Name}
 		}
```

Comment:
Any idea on whether returning an error that will likely propagate to an individual InstallPlan resource's status (if I understand the code correctly) is sufficient in the context of visibility back to an admin/user/etc.? Do we also emit events for failed InstallPlans?

Reply:
I can double check but I believe we do.

We need to be careful around emitting events though, since we now retry installplans (and do not have an explicit list of errors which are terminal IIRC) we may end up shooting off a bunch of events for one installplan failing. @benluddy would know more.